project/src/test/java/com/google/sps/data/PlacesFetcherTest.java

```diff
@@ -125,13 +131,21 @@ public void fetch_zeroSearchResults_returnsEmptyList() throws Exception {
   }
 
   @Test
-  public void fetch_validSearchResults_returnsListOfPlaces()
-      throws ApiException, InterruptedException, IOException {
+  public void fetch_validSearchResults_returnsListOfPlaces() throws Exception {
     PlacesFetcher spiedFetcher = spy(placesFetcher);
     doReturn(PLACE_DETAILS_1).doReturn(PLACE_DETAILS_2)
         .when(spiedFetcher).getPlaceDetails(any(PlaceDetailsRequest.class));
     doReturn(SEARCH_RESULT_ARR)
         .when(spiedFetcher).getPlacesSearchResults(any(TextSearchRequest.class));
     assertEquals(ImmutableList.of(PLACE_1, PLACE_2), spiedFetcher.fetch());
   }
+
+  @Test
+  public void fetch_PlaceDetailsQueryFails_throwsFetcherException()
+      throws FetcherException {
+    PlacesFetcher spiedFetcher = spy(placesFetcher);
+    doReturn(new PlacesSearchResult[] {SEARCH_RESULT_INVALID})
```

Comment:
I don't really understand how this works. 
What makes SEARCH_RESULT_INVALID to be actually invalid? What makes `getPlacesSearchResults` throw an exception? 
Why not call `doThrow` for some API exception, and then verify that `FetcherException` was actually thrown up the stack?

Reply:
SEARCH_RESULT_INVALID is invalid because it has an invalid placeID. 

I couldn't find a way to mock or change a TextSearchRequest or PlaceDetailsRequest or the CONTEXT to be invalid so this is the best way I found to generate an invalid request and check that querying it throws FetcherException.

Regarding the use of doThrow - since the method that is supposed to throw it is await() and I cannot stub it, I didn't find it useful.

I don't think the test I have now is good but after trying several things it is the best I came up with so I will be happy to discuss it f2f.

Reply:
I think something's missing here. Is the test making an actual call to the API? 
Let's talk F2F about this.

Reply:
Fixed this as we discussed. Thanks!